ngraph/core/src/op/util/arithmetic_reduction.cpp

```diff
@@ -46,24 +46,27 @@ void op::util::ArithmeticReduction::set_reduction_axes(const AxisSet& reduction_
             ->output(0));
 }
 
-void op::util::ArithmeticReduction::validate_and_infer_types()
+PartialShape op::util::ArithmeticReduction::infer_reduction_output_shape(const bool keep_dims)
 {
-    NGRAPH_OP_SCOPE(util_ArithmeticReduction_validate_and_infer_types);
-    auto input_shape = get_input_partial_shape(0);
-    const auto input_rank = input_shape.rank();
+    const PartialShape& data_ps = get_input_partial_shape(0);
+    PartialShape result_ps{PartialShape::dynamic()};
+    Rank data_rank = data_ps.rank();
 
-    PartialShape result_shape{PartialShape::dynamic()};
+    if (data_rank.is_static() && keep_dims)
+    {
+        result_ps = PartialShape::dynamic(data_rank);
+    }
 
-    auto axes = get_constant_from_source(input_value(1));
-    if (input_rank.is_static() && axes)
+    const auto& axes = get_constant_from_source(this->input_value(1));
```

Comment:
nitpick: `this->` is redundant 😄 

Maybe we can use one common `if` for static rank:
```
if (data_rank.is_static()) {
if (keep_dims) {}
if (axes) {}
}
```